java/java.editor.base/src/org/netbeans/modules/java/editor/base/semantic/Utilities.java

```diff
@@ -557,7 +557,7 @@ private static int findBodyStartImpl(CompilationInfo info, Tree cltree, Compilat
     
     public static int findBodyStart(final CompilationInfo info, final Tree cltree, final CompilationUnitTree cu, final SourcePositions positions, final Document doc) {
         Kind kind = cltree.getKind();
-        if (!TreeUtilities.CLASS_TREE_KINDS.contains(kind) && kind != Kind.METHOD)
+        if (!TreeUtilities.CLASS_TREE_KINDS.contains(kind) && kind != Kind.METHOD && !cltree.getKind().toString().equals("RECORD"))
```

Comment:
Unless this was tried already, and rejected for some reason, I would recommend to add Tree.Kind.RECORD into TreeUtilities.CLASS_TREE_KINDS. The purpose of that constant is to keep all kinds that map to ClassTree.

Reply:
Tree.Kind.RECORD is a final Enumset, So adding RECORD here as a preview feature would result a compilation failure for the versions below 14.